HLTrigger/Timer/plugins/FastTimerServiceClient.cc

```diff
@@ -223,12 +243,119 @@ FastTimerServiceClient::fillPathSummaryPlots(DQMStore::IBooker & booker, DQMStor
+  // if no ME available, return
+  if ( !lumiVsLS ) {
+    edm::LogWarning("FastTimerServiceClient") << "no " << name << " ME is available in " << folder << std::endl;
+    return;
+  }
+
+  // get range and binning for new MEs x-axis
+  size_t size        = lumiVsLS->getTProfile()->GetXaxis()->GetNbins();
+  //  double xmin        = lumiVsLS->getTProfile()->GetMinimum();
+  //  double xmax        = lumiVsLS->getTProfile()->GetMaximum()*1.1;
+  std::string xtitle = lumiVsLS->getTProfile()->GetYaxis()->GetTitle();
+
+  std::vector<double> lumi;
+  std::vector<int> LS;
+  for ( size_t ibin=1; ibin <= size; ++ibin ) {
+    // avoid to store points w/ no info
+    if ( lumiVsLS->getTProfile()->GetBinContent(ibin) == 0. ) continue;
+
+    lumi.push_back( lumiVsLS->getTProfile()->GetBinContent(ibin) );
+    LS.push_back  ( lumiVsLS->getTProfile()->GetXaxis()->GetBinCenter(ibin) );
+  }
+
+  booker.setCurrentFolder(current_path);
+  getter.setCurrentFolder(current_path);
+  for ( auto m : menames ) {
+    std::string label = m;
+    label.erase(label.find("_byls"));
+
+    MonitorElement* me = getter.get(current_path + "/" + m);
+    double ymin        = me->getTProfile()->GetMinimum();
+    double ymax        = me->getTProfile()->GetMaximum();
+    std::string ytitle = me->getTProfile()->GetYaxis()->GetTitle();
+
+    MonitorElement* meVsLumi = getter.get( current_path + "/" + label + "_" + suffix );
+    if (meVsLumi) {
+      assert( meVsLumi->getTProfile()->GetXaxis()->GetXmin()  == xmin );
+      assert( meVsLumi->getTProfile()->GetXaxis()->GetXmax()  == xmax );
+      meVsLumi->Reset(); // do I have to do it ?!?!?
+    } else {
+      meVsLumi = booker.bookProfile(label + "_" + suffix, label + "_" + suffix, nbins, xmin, xmax, ymin, ymax);
+      //    TProfile* meVsLumi_p = meVsLumi->getTProfile();
+      meVsLumi->getTProfile()->GetXaxis()->SetTitle(xtitle.c_str());
+      meVsLumi->getTProfile()->GetYaxis()->SetTitle(ytitle.c_str());
+    }
+    for ( size_t ils=0; ils < LS.size(); ++ils ) {
+      int ibin = me->getTProfile()->GetXaxis()->FindBin(LS[ils]);
+      double y = me->getTProfile()->GetBinContent(ibin);
+
+      meVsLumi->Fill(lumi[ils],y);
+    }
+  }
+  
+  
+}
+
+void 
+FastTimerServiceClient::fillLumiMePSetDescription(edm::ParameterSetDescription & pset) {
+  pset.add<std::string>("folder", "HLT/LumiMonitoring");
+  pset.add<std::string>("name"  , "lumiVsLS");
+  pset.add<int>   ("nbins",  6500 );
+  pset.add<double>("xmin",      0.);
+  pset.add<double>("xmax",  13000.);
+}
+
+
 void
 FastTimerServiceClient::fillDescriptions(edm::ConfigurationDescriptions & descriptions) {
   // The following says we do not know what parameters are allowed so do no validation
   // Please change this to state exactly what you do use, even if it is no parameters
   edm::ParameterSetDescription desc;
   desc.addUntracked<std::string>( "dqmPath", "HLT/TimerService");
+  desc.add<bool>( "doPlotsVsSCALlumi",  true  );
```

Comment:
can use consistently use `scalLumi` or `...ScalLumi` and `pixelLumi` or `...PixelLumi` for all the variables and parameter names ?